--- ./libdmapsharing/Makefile.am.link	2012-01-02 10:04:15.046243310 +0100
+++ ./libdmapsharing/Makefile.am	2012-01-02 10:04:41.236928142 +0100
@@ -57,14 +57,16 @@
 	$(MDNS_CFLAGS)
 
 libdmapsharing_3_0_la_LDFLAGS = \
-	-version-info @SO_VERSION@ \
+	-version-info @SO_VERSION@
+
+libdmapsharing_3_0_la_LIBADD = \
 	$(GLIB_LIBS) \
 	$(GTHREAD_LIBS) \
 	$(GOBJECT_LIBS) \
 	$(GDKPIXBUF_LIBS) \
 	$(MDNS_LIBS) \
 	$(GSTREAMERAPP_LIBS) \
-	$(SOUP_LIBS)
+	$(SOUP_LIBS) -lm
 
 libdmapsharingincludedir = \
 	$(includedir)/libdmapsharing-@API_VERSION@/libdmapsharing
